kubejs/server_scripts/enigmatica/kubejs/normal/recipetypes/astralsorcery/block_transmutation.js

```diff
@@ -2,16 +2,30 @@ onEvent('recipes', (event) => {
     if (global.isNormalMode == false) {
         return;
     }
+
+    const id_prefix = 'enigmatica:normal/astralsorcery/block_transmutation/';
+
     var data = {
         recipes: [
-            { inputTag: '#forge:ores/iron', output: 'astralsorcery:starmetal_ore', starlight: 100 },
-            { inputTag: '#forge:workbenches', output: 'astralsorcery:altar_discovery', starlight: 60 }
+            {
+                inputTag: '#forge:ores/iron',
+                output: 'astralsorcery:starmetal_ore',
+                starlight: 100,
+                id: `${id_prefix}starmetal_ore_from_`
```

Comment:
basically to pinpoint the generated id to one of those recipes, instead of merely originating in that file:
```
${id_prefix}starmetal_ore_from_md5_fw7678a76f7f6w78fw787f
${id_prefix}starmetal_ore_from_<name of the input iron ore block>
```
```
${id_prefix}luminous_crafting_table_from_md5_8765d4gfh5g6d7sds8f7
${id_prefix}luminous_crafting_table_from_<name of the wood's crafting table variant>
```

Reply:
I don't understand that ^^' where is that done?